### PR TITLE
Sampling rate overhaul

### PR DESCRIPTION
**Pull request by cachedout**
## What does this PR do?

1. Introduces a new CLI flag called `--drop-unsampled`. When this flag is passed in, `apm-server.sampling.keep_unsampled` is set to `true` on the APM server.
2. Introduces new flags for each Opbean (`--opbeans-$lang-sample-rate`) which controls the sampling rate for each opbean.
3. Various small fixups encountered along the way.
## Why is it important?
Requested by APM team.

## Related issues
Closes https://github.com/elastic/apm-integration-testing/issues/1241

### CODE REVIEW

File: docker/dyno/tests/conftest.py [hunk 1]
```diff
@@ -19,7 +19,10 @@
 """
 Tests for Dyno control plane
 """
-import toxiproxy
+try:
+    import toxiproxy
```
Comment (cachedout) [inline docker/dyno/tests/conftest.py:23]: This just cleans up some errors that can happen with pytest during test discovery.

**Comment by sqren** [comment]
This is awesome, thanks! Can we make it the default to drop unsampled, and have a sampling rate less than 100%?

**Comment by cachedout** [comment]
Hi @sqren 

Here is the logic for deciding how to set the sampling flag: https://github.com/elastic/apm-integration-testing/pull/1255/files#diff-df11512a4602ab3dcd2e68f8f85f7462684d54050bfcb70ba7f94b39d53bea8aR203

The idea here was to set this as the default if it was set explicitly or the version running is at least 7.16 [as suggested here](https://github.com/elastic/apm-integration-testing/issues/1241#issuecomment-909627150).

Are you saying you'd like dropping un-sampled transactions to be the default behavior regardless of version?

**Comment by sqren** [comment]
> The idea here was to set this as the default if it was set explicitly or the version running is at least 7.16 as suggested here.

That sounds great, thanks. I missed the point about 7.16.

So running `./scripts/compose.py start master` will drop unsampled transactions and have a sampling rate of 10%?

**Comment by cachedout** [comment]
> @sqren  So running ./scripts/compose.py start master will drop unsampled transactions and have a sampling rate of 10%?

That is correct.

**Review by sqren — APPROVED** [review]
lgtm

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**